File: database/app/src/main/java/com/google/firebase/referencecode/database/ShardingActivity.java
package com.google.firebase.referencecode.database;

import androidx.appcompat.app.AppCompatActivity;

import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.FirebaseDatabase;

public class ShardingActivity extends AppCompatActivity {

    private void getMultipleInstances() {
        // [START rtdb_multi_instance]
        // Get the default database instance for an app
        DatabaseReference primary = FirebaseDatabase.getInstance()
                .getReference();

        // Get a secondary database instance by URL
        DatabaseReference secondary = FirebaseDatabase.getInstance("https://testapp-1234.firebaseio.com")
                .getReference();
        // [END rtdb_multi_instance]
    }
}

File: database/app/src/main/java/com/google/firebase/referencecode/database/kotlin/ShardingActivity.kt
package com.google.firebase.referencecode.database.kotlin

import androidx.appcompat.app.AppCompatActivity
import com.google.firebase.database.ktx.database
import com.google.firebase.ktx.Firebase

class ShardingActivity : AppCompatActivity() {

    private fun getMultipleInstances() {
        // [START rtdb_multi_instance]
        // Get the default database instance for an app
        val primary = Firebase.database.reference

        // Get a secondary database instance by URL
        val secondary = Firebase.database("https://testapp-1234.firebaseio.com").reference
        // [END rtdb_multi_instance]
    }
}
